faker/providers/internet/__init__.py

```diff
@@ -99,15 +102,39 @@ def url(self):
         pattern = self.random_element(self.url_formats)
         return self.generator.parse(pattern)
 
-    def ipv4(self):
+    def ipv4(self, network=False):
+        """ Produce a random IPv4 address or network with a valid CIDR.
+
+        IP code generation inspired by ipaddress module: https://github.com/phihag/ipaddress
```

Comment:
If `phihag/ipaddress` and Python3's `ipaddress` have the same api and functionality, I think i'd rather add `phihag/ipaddress` as a requirement in `setup.py` for Python < 3.3.